beacon-chain/rpc/eth/helpers/sync.go

```diff
@@ -35,3 +37,21 @@ func ValidateSync(ctx context.Context, syncChecker sync.Checker, headFetcher blo
 	}
 	return status.Error(codes.Unavailable, "Syncing to latest head, not ready to respond")
 }
+
+func IsOptimistic(ctx context.Context, st state.BeaconState, headFetcher blockchain.HeadFetcher) (bool, error) {
+	root, err := st.HashTreeRoot(ctx)
+	if err != nil {
+		return false, errors.Wrap(err, "could not get state root")
+	}
+	header := st.LatestBlockHeader()
+	header.StateRoot = root[:]
+	headRoot, err := header.HashTreeRoot()
```

Comment:
This helper function is being called in many RPC points and computes 2 HTRs. One is most probably cached. Can we not get the block's root from the state summary or similar?